src/todos/todos.database.service.ts

```diff
@@ -1,10 +1,56 @@
-import { Injectable, NotFoundException } from '@nestjs/common'
-import { initialTodos } from './mocks'
+import {
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+  OnModuleInit,
+  OnModuleDestroy,
+} from '@nestjs/common'
+import * as path from 'path'
+import * as fs from 'fs/promises'
+
 import type { TodoItem, Todos } from 'types/todos'
 
 @Injectable()
-export class TodosDatabaseService {
-  static todos: Todos = initialTodos
+export class TodosDatabaseService implements OnModuleInit, OnModuleDestroy {
+  private static todos: Todos = new Map()
+
+  private readonly todosFilePath: string = path.resolve(
+    __dirname,
+    '../../../todos.json',
+  )
```

Comment:
TBD if it's a correct way to resolve pathes

Reply:
It's correct; one of the approaches.